### ISSUE TITLE
[mdoc] Error when updating on a .NET Core assembly

### ISSUE THREAD

**Issue by joelmartinez** [mono/api-doc-tools#5]
In the currently released version of mdoc, you receive the following error when you try to `mdoc update` a .NET Core assembly:

```bash
$ mdoc update --debug bin/Debug/netcoreapp1.0/dnnew.dll -o en -L /usr/local/share/dotnet/shared/Microsoft.NETCore.App/1.0.0-rc2-3002702/
mdoc: System.NotSupportedException: Specified method is not supported.
  at Mono.Cecil.PE.ImageReader.ReadArchitecture () [0x00040] in <6f5dcfb43b83409e9bb6016a08899100>:0 
  at Mono.Cecil.PE.ImageReader.ReadImage () [0x0005c] in <6f5dcfb43b83409e9bb6016a08899100>:0 
  at Mono.Cecil.PE.ImageReader.ReadImage (Mono.Disposable`1[T] stream, System.String file_name) [0x00008] in <6f5dcfb43b83409e9bb6016a08899100>:0 
  at Mono.Cecil.ModuleDefinition.ReadModule (Mono.Disposable`1[T] stream, System.String fileName, Mono.Cecil.ReaderParameters parameters) [0x00006] in <6f5dcfb43b83409e9bb6016a08899100>:0 
  at Mono.Cecil.ModuleDefinition.ReadModule (System.String fileName, Mono.Cecil.ReaderParameters parameters) [0x0006f] in <6f5dcfb43b83409e9bb6016a08899100>:0 
```

It looks like we should be able to work around this with a newer version of Mono.Cecil (or will be soon as they are obviously still adding some facilities like a [default AssemblyResolver](https://github.com/jbevain/cecil/issues/306) for .net core).

**Comment by joelmartinez** [comment]
Now that we're using the latest mono.cecil, it's able to read the assembly ... doing an `mdoc update` on a .net core assembly results in the following error:

```
mdoc: Mono.Cecil.AssemblyResolutionException: Failed to resolve assembly: 'System.Runtime, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
  at Mono.Cecil.BaseAssemblyResolver.Resolve (Mono.Cecil.AssemblyNameReference name, Mono.Cecil.ReaderParameters parameters) [0x0011d] in <6e8b97570af94606848d36e32f55026d>:0 
  at Mono.Cecil.BaseAssemblyResolver.Resolve (Mono.Cecil.AssemblyNameReference name) [0x00007] in <6e8b97570af94606848d36e32f55026d>:0 
  at Mono.Cecil.DefaultAssemblyResolver.Resolve (Mono.Cecil.AssemblyNameReference name) [0x0002b] in <6e8b97570af94606848d36e32f55026d>:0 
  at Mono.Cecil.MetadataResolver.Resolve (Mono.Cecil.TypeReference type) [0x00048] in <6e8b97570af94606848d36e32f55026d>:0 
  at Mono.Cecil.ModuleDefinition.Resolve (Mono.Cecil.TypeReference type) [0x00006] in <6e8b97570af94606848d36e32f55026d>:0 
  at Mono.Cecil.TypeReference.Resolve () [0x00013] in <6e8b97570af94606848d36e32f55026d>:0 
  at Mono.Cecil.Mixin.CheckedResolve (Mono.Cecil.TypeReference self) [0x00000] in <6e8b97570af94606848d36e32f55026d>:0 
  at Mono.Cecil.SignatureReader.ReadCustomAttributeEnum (Mono.Cecil.TypeReference enum_type) [0x00000] in <6e8b97570af94606848d36e32f55026d>:0 
  at Mono.Cecil.SignatureReader.ReadCustomAttributeElementValue (Mono.Cecil.TypeReference type) [0x0003d] in <6e8b97570af94606848d36e32f55026d>:0 
  at Mono.Cecil.SignatureReader.ReadCustomAttributeElement (Mono.Cecil.TypeReference type) [0x00032] in <6e8b97570af94606848d36e32f55026d>:0 
  at Mono.Cecil.SignatureReader.ReadCustomAttributeFixedArgument (Mono.Cecil.TypeReference type) [0x00018] in <6e8b97570af94606848d36e32f55026d>:0 
  at Mono.Cecil.SignatureReader.ReadCustomAttributeConstructorArguments (Mono.Cecil.CustomAttribute attribute, Mono.Collections.Generic.Collection`1[T] parameters) [0x00034] in <6e8b97570af94606848d36e32f55026d>:0 
  at Mono.Cecil.MetadataReader.ReadCustomAttributeSignature (Mono.Cecil.CustomAttribute attribute) [0x00045] in <6e8b97570af94606848d36e32f55026d>:0 
  at Mono.Cecil.CustomAttribute.<Resolve>m__1 (Mono.Cecil.CustomAttribute attribute, Mono.Cecil.MetadataReader reader) [0x00000] in <6e8b97570af94606848d36e32f55026d>:0 
  at Mono.Cecil.ModuleDefinition.Read[TItem,TRet] (TItem item, System.Func`3[T1,T2,TResult] read) [0x00029] in <6e8b97570af94606848d36e32f55026d>:0 
  at Mono.Cecil.CustomAttribute.Resolve () [0x0001d] in <6e8b97570af94606848d36e32f55026d>:0 
  at Mono.Cecil.CustomAttribute.get_ConstructorArguments () [0x00000] in <6e8b97570af94606848d36e32f55026d>:0 
  at Mono.Documentation.MDocUpdater+<GetCustomAttributes>c__Iterator0.MoveNext () [0x0013b] in <4e7920fba2c640f6b210a3ac0e6e72e1>:0 
  at System.Linq.Enumerable.Any[TSource] (System.Collections.Generic.IEnumerable`1[T] source) [0x00018] in <de06f25147c848e79f091ac5f34e9b67>:0 
  at Mono.Documentation.MDocUpdater.MakeAttributes (System.Xml.XmlElement root, System.Collections.Generic.IEnumerable`1[T] attributes, Mono.Cecil.TypeReference t) [0x00000] in <4e7920fba2c640f6b210a3ac0e6e72e1>:0 
  at Mono.Documentation.MDocUpdater.AddIndexAssembly (Mono.Cecil.AssemblyDefinition assembly, System.Xml.XmlElement parent) [0x00162] in <4e7920fba2c640f6b210a3ac0e6e72e1>:0 
  at Mono.Documentation.MDocUpdater.DoUpdateAssemblies (System.String source, System.String dest) [0x0011b] in <4e7920fba2c640f6b210a3ac0e6e72e1>:0 
  at Mono.Documentation.MDocUpdater.Run (System.Collections.Generic.IEnumerable`1[T] args) [0x00444] in <4e7920fba2c640f6b210a3ac0e6e72e1>:0 
  at Mono.Documentation.MDoc.Run (System.String[] args) [0x001f6] in <4e7920fba2c640f6b210a3ac0e6e72e1>:0 
  at Mono.Documentation.MDoc.Main (System.String[] args) [0x00006] in <4e7920fba2c640f6b210a3ac0e6e72e1>:0 
```

**Comment by joelmartinez** [comment]
Investigating the solution to this ... filed a GH Issue on Mono.Cecil: https://github.com/jbevain/cecil/issues/329

**Comment by joelmartinez** [comment]
This issue was inadvertently resolved when I merged [PR#18](https://github.com/mono/api-doc-tools/pull/18) ... which brought in https://github.com/jbevain/cecil/commit/bb40c2108ecf303691d0536c4f9d3b9035790c5c

So .NET core assemblies can now be documented, *as long as you include a reference to the lookup directory* (platform dependent, obviously). If you're documenting a single assembly, you can use `-L`:

    mdoc update -o some/corelib.dll -L /usr/local/share/dotnet/shared/Microsoft.NETCore.App/1.1.0/

Or under frameworks mode:

```xml
<?xml version="1.0" encoding="utf-8"?>
<Frameworks>
  <Framework Name="Some .NET Core Framework" Source="path/to/acorelib">
    <assemblySearchPath>/usr/local/share/dotnet/shared/Microsoft.NETCore.App/1.1.0/</assemblySearchPath>
  </Framework>
</Frameworks>
```

/cc @dend
